coral/src/domain/auth-user/auth-user-api.ts

```diff
@@ -15,7 +15,31 @@ const getAuthUserMockForLogin = (
   return api.post("/login" as keyof ApiPaths, userLogin);
 };
 
+// user roles are added dynamically by admins, we can't make them enums
+// but "SUPERADMIN" is the default role set by us. It's highly unlikely
+// that users would change that string, so we can use it to identify
+// the superadmin role, which has different view-access
+const SUPERADMIN_USERROLE = "SUPERADMIN";
+function isSuperAdmin(user: AuthUser) {
```

Comment:
This looks like an typeguard, so lets make it a proper one. 

```
function isSuperAdmin(user: AuthUser): AuthUser & {userrole: SUPERADMIN_USERROLE} {}
```

Or by using [typefest](https://github.com/sindresorhus/type-fest)

```
function isSuperAdmin(user: AuthUser): OverrideProperties<AuthUser, {userrole: SUPERADMIN_USERROLE}> {}
```

Reply:
Does it count as typeguard even if SUPERADMIN is not REALLY a type? 

`isSuperAdmin` currently returns a boolean, how would you use it in our context?  (I've not worked with typeguards a lot)

Reply:
My example was not complete. I missed the `user is ...` from the return type. You can also define the type separately

```
type SuperUser = AuthUser & { userrole: SUPERADMIN_USERROLE }
function isSuperAdmin(user: AuthUser): user is SuperUser {}
```

Reply:
Thank you 🎉  
Updated! https://github.com/Aiven-Open/klaw/pull/1587/commits/ed1ee8c52751b3dedb16a05c1c0786dca512c2e5